libs/sync/closer_test.go

```diff
@@ -0,0 +1,17 @@
+package sync_test
+
+import (
+	"testing"
+
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+)
+
+func TestCloser(t *testing.T) {
+	closer := tmsync.NewCloser()
+
+	for i := 0; i < 10; i++ {
+		closer.Close()
+	}
+
+	<-closer.Done()
```

Comment:
Should test that `<-Done()` blocks before we call `Close()`.

Reply:
Good call.